hashicorp/vault
# Remove the Bulma CSS Framework

This PR removes bulma imports from our package.json (the "easy" part). The rest of the PR adds back in the missing styles we relied on bulma for.

**Noteable Changes:**
* `$tablet` was the exact same as the var `$mobile`—$tablet originated from bulma. I replaced all use cases of `$tablet` with `$mobile`. This makes up a lot of changes, but there was more use cases of mobile than tablet.
![image](https://user-images.githubusercontent.com/6618863/233187803-86675d99-1d9c-4df4-bbc2-5980600b65c7.png)

* I removed a good chunk of older vars. Things like the `$vault-xx` colors making it clear that we now only use `$ui-xx` colors.
* I replaced the old `helpers.scss` file and made a directory called "helper-classes". Based on a RFC discussion I added the following helper files: `colors.scss`, `flexbox-and-grid.scss`, `layout.scss`, `general.scss`, `spacing.scss`, `typography.scss`.
* Cleaned up our core/xx.scss files, which meant first merging in relevant bulma classes and then clarifying the class(es) defined in each file. In some cases, like the `form.scss` it meant deleting this file and breaking up the styling into different core/xx.scss files.
* When it made sense, I removed component css and replaced with a helper. I tried to keep these to a minimum. 

**Scope Note:**
* This pr has a narrow scope: add back the missing bulma classes. I did my best to keep to this scope. That means messy parts of our css remain. Thus if you see something like an `important!` I didn't add it. It was either from bulma or was like that before and you're seeing it because I moved the css block to another file. Same with hardcoded size and color vars. **The key here: 95% of the time I didn't write new css. I merged css and moved it around.** 

## QA steps:
_Same steps are on the internal google doc._
1. Setup a binary from main. To do this, on vault-enterprise (remember to pull for latest changes) run `make static-dist` and then `make entdev-ui`. Start up the instance using `vrd` and then navigate to  `http://localhost:8200/ui/`. This will be the comparison branch.
2. Setup a local instance of vault running this sidebranch. To do this, you'll run a local environment like you normally would but use `vrd2` and `yarn start2` instead. You'll access this instance at `http://localhost:4202/ui/`. Remember to yarn install!

You'll have two windows that you can now compare.
![image](https://user-images.githubusercontent.com/6618863/232892070-7a92a1dd-851e-4f9d-bc76-206a16b494bb.png)

3. **Investigate what is most familiar to you.** Are you familiar with the `autocomplete-input`, or maybe you know how to easily play around with the charts, then check those things out. I know replication well and understand it can be hard to go through the full flow, so that's one area I QA'd heavily. I don't expect anyone else to check what a DR secondary cluster looks like.

5. Remember the app reuses a lot of the same styling. So think about the app in chunks of components and classes. Do you know this form uses tables? Great, play with that part of the form. Do you need to look at every input? No, but maybe look at a regular input, a maskedInput and a textarea input?

6. If you spot a styling difference, first look at the Dev console and compare the css. The binary and your local version will not be exactly the same but you should see very similar styling. Here's a screenshot of what I mean.
![image](https://user-images.githubusercontent.com/6618863/232871480-85f5694c-def4-44a0-953c-3922742d6050.png)

**If you spot a bug**, report it on the internal Google doc. 

### Known differences 
* There are some very slight color differences in a very few cases (mostly in the AlertBanners). If you have a color question and it's not an eye-catching difference, it's likely intentional. 
* `.toolbar-link:hover` works in this branch. It is overridden in the binary. 
![image](https://user-images.githubusercontent.com/6618863/233100960-9d2833f3-b17a-4a3c-8f1e-5fcc18c56e66.png)
* The "x" on the modal is different on purpose. Before we used a custom icon made by bulma. In this branch we're using a flight-icon. 
![image](https://user-images.githubusercontent.com/6618863/233101355-aa91e3c7-61e5-4151-9a4f-f84b56dc4dfd.png)
* The active tab on auth mounts and transit key actions have been fixed on this branch while they're still broken on main. 
![image](https://user-images.githubusercontent.com/6618863/233158809-32a8a20a-a534-4755-9652-37ee5847ab82.png)
* Likely elsewhere but on the binary there is an issue with the is-active status on the calendar widget. It correctly shows as blue text in this sidebranch. 
![image](https://user-images.githubusercontent.com/6618863/233407555-7a1ebc20-e783-4fac-9c57-cbe16894ecb1.png)
* This branch fixes an issue again with active tabs. This one shows on Tools -> Unwrap data (you need to unwrap) -> Data 
![image](https://user-images.githubusercontent.com/6618863/233411882-159670e3-5c76-404b-b9f8-c1de28dbdf7a.png)
